arch/xtensa/src/esp32/esp32_allocateheap.c

```diff
@@ -93,7 +93,7 @@ void up_allocate_heap(FAR void **heap_start, size_t *heap_size)
  *
  ****************************************************************************/
 
-#if CONFIG_MM_REGIONS > 1
+#if CONFIG_MM_REGIONS > 1 && defined(CONFIG_ESP32_SPIRAM)
```

Comment:
```suggestion
#if CONFIG_MM_REGIONS > 1
```

This is what's breaking the build.
This condition should be moved to the line bellow where the PSRAM region is actually added.